pkg/trimaran/targetloadpacking/targetloadpacking.go

```diff
@@ -0,0 +1,257 @@
+	err := updateArguments(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	podAssignEventHandler := trimaran.New()
+	pl := &TargetLoadPacking{
+		handle: handle,
+		client: http.Client{
+			Timeout: httpClientTimeoutSeconds,
+		},
+		eventHandler: podAssignEventHandler,
+	}
+
+	pl.handle.SharedInformerFactory().Core().V1().Pods().Informer().AddEventHandler(podAssignEventHandler)
+
+	// populate metrics before returning
+	err = pl.updateMetrics()
+	if err != nil {
+		klog.Warningf("unable to populate metrics initially: %v", err)
+	}
+	go func() {
+		metricsUpdaterTicker := time.NewTicker(time.Second * metricsUpdateIntervalSeconds)
+		for range metricsUpdaterTicker.C {
+			err = pl.updateMetrics()
+			if err != nil {
+				klog.Warningf("unable to update metrics: %v", err)
+			}
+		}
+	}()
+
+	return pl, nil
+}
+
+func (pl *TargetLoadPacking) updateMetrics() error {
+	req, err := http.NewRequest(http.MethodGet, watcherAddress+WatcherBaseUrl, nil)
+	if err != nil {
+		klog.Errorf("new watcher request failed: %v", err)
+	}
+	klog.Info("watcher request successful")
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := pl.client.Do(req) //TODO(aqadeer): Add a couple of retries for transient errors
+	if err != nil {
+		klog.Errorf("request to watcher failed: %v", err)
+		return err
+	}
+	defer resp.Body.Close()
+	klog.V(6).Infof("received status code %v from watcher", resp.StatusCode)
+	if resp.StatusCode == http.StatusOK {
+		data := watcher.Data{NodeMetricsMap: make(map[string]watcher.NodeMetrics)}
+		var metrics = watcher.WatcherMetrics{Data: data}
+		dec := gojay.BorrowDecoder(resp.Body)
+		defer dec.Release()
+		err = dec.Decode(&metrics)
+		if err != nil {
+			klog.Errorf("unable to decode watcher metrics: %v", err)
+		}
+		pl.metrics = metrics
+	} else {
+		klog.Errorf("received status code %v from watcher", resp.StatusCode)
+	}
+	return nil
+}
+
+func (pl *TargetLoadPacking) Name() string {
+	return Name
+}
+
+func getArgs(obj runtime.Object) (*pluginConfig.TargetLoadPackingArgs, error) {
+	if obj == nil {
+		return &pluginConfig.TargetLoadPackingArgs{
+			TargetCPUUtilization: defaultHostCPUThresholdPercent,
+			DefaultCPURequests:   defaultRequestsMilliCores,
+		}, nil
+	}
+	if targetLoadPackingArgs, ok := obj.(*pluginConfig.TargetLoadPackingArgs); ok {
+		return targetLoadPackingArgs, nil
+	}
+	return nil, fmt.Errorf("want args to be of type TargetLoadPackingArgs, got %T", obj)
+}
+
+func (pl *TargetLoadPacking) Score(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
+	nodeInfo, err := pl.handle.SnapshotSharedLister().NodeInfos().Get(nodeName)
+	if err != nil {
+		return 0, framework.NewStatus(framework.Error, fmt.Sprintf("getting node %q from Snapshot: %v", nodeName, err))
+	}
+
+	metrics := pl.metrics                                    // copy to maintain snapshot lest updateMetrics() updates the value
```

Comment:
You can read explanation for no snapshot in PreScore [here](https://github.com/kubernetes-sigs/scheduler-plugins/pull/61#issuecomment-709636411) 

I wanted to avoid lock for this block of code, so a local function snapshot is okay.